homeassistant/components/openhome/update.py

```diff
@@ -0,0 +1,128 @@
+"""Update entities for Linn devices."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.components.update import (
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entities for Reolink component."""
+
+    _LOGGER.debug("Setting up config entry: %s", config_entry.unique_id)
+
+    device = hass.data[DOMAIN][config_entry.entry_id]
+
+    entity = OpenhomeUpdateEntity(device)
+
+    await entity.async_update()
+
+    async_add_entities([entity])
+
+
+class OpenhomeUpdateEntity(UpdateEntity):
+    """Update entity for a Linn DS device."""
+
+    def __init__(self, device):
+        """Initialize a Linn DS update entity."""
+        self._attr_device_class = UpdateDeviceClass.FIRMWARE
+        self._attr_supported_features = UpdateEntityFeature.INSTALL
+        self._name = None
+        self._device = device
+        self._installed_version = None
+        self._latest_version = None
+        self._release_summary = None
+        self._release_url = None
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return f"{self._device.uuid()}-update"
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return f"{self._name} Firmware Update"
+
+    @property
+    def installed_version(self) -> str | None:
+        """Version currently in use."""
+        return self._installed_version
+
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        return self._latest_version
+
+    @property
+    def release_summary(self) -> str | None:
+        """Description of the latest firmware release."""
+        return self._release_summary
+
+    @property
+    def release_url(self) -> str | None:
+        """URL to the full release notes of the latest version available."""
+        return self._release_url
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return DeviceInfo(
+            identifiers={
+                (DOMAIN, self._device.uuid()),
+            },
+            manufacturer=self._device.device.manufacturer,
+            model=self._device.device.model_name,
+            name=self._device.device.friendly_name,
+        )
+
+    async def async_update(self) -> None:
+        """Update state of entity."""
+
+        self._name = await self._device.room()
```

Comment:
This should not be needed. Instead, set `self._attr_has_entity_name = True` and Home Assistant handles the device name being added to the name automatically.

Reply:
Much more elegant. I didn't know this (not mentioned on https://developers.home-assistant.io/docs/core/entity/update)

Reply:
It's a generic entity feature that all entities have 👍 

If you add a device class to the entity, you don't even need to give it a name and we offer translated ones.